app/validators/date_range_validator.rb

```diff
@@ -1,7 +1,17 @@
 class DateRangeValidator < ActiveModel::EachValidator
+  def initialize(**kwargs)
+    super(**kwargs)
+  end
+
   def validate_each(record, attribute, value)
-    unless range.cover?(value)
-      record.errors.add(attribute, :date_range, start_date: start_date_str, end_date: end_date_str)
+    unless value.nil?
```

Comment:
This level of indenting doesn't read nicely.

You could simplify this using guard clauses to return early under certain conditions